### PR TITLE
Makefile: Respect ARFLAGS and refactor OSX ar test

### PR DESCRIPTION
**Pull request by gebart**
I ran into a problem when trying to use LLVM `llvm-ar` instead of GNU binutils `ar` for building static archives:

```
llvm-ar: Unknown command line argument '-rcs'.  Try: 'llvm-ar -help'
llvm-ar: Did you mean '-mlsm'?
```

It works if I remove the leading dash on the `ar` command line. I also found that there is a check for OSX ar in Makefile.base since #1333, #1334 which gave a false positive because of the same problem, the command errored out because of wrong arguments, not because of unsupported features.

I have refactored the root Makefiles to use a new variable `AROPTS` for passing options to `ar`.

I need someone who has access to the broken version of OSX `ar` (a version for which the `dist/ar-wrapper` script was written) to test that the check is still valid after removing the leading dash on the arguments to `ar` (@thomaseichinger ?)

### CODE REVIEW

**Comment by gebart** [comment]
Updated to use the name `ARFLAGS` instead of `AROPTS`.

**Comment by thomaseichinger** [comment]
@gebart What exactly should I test? These changes don't break building for `native`, `ar-wrapper` gets called as in `master`.

**Comment by gebart** [comment]
@thomaseichinger I wanted to verify that the `ar-wrapper` is still called on OSX, but not on GNU or LLVM.

Side note: I thought OSX was using LLVM/Clang for its toolchain, but I guess they don't use llvm-ar at least.

**Comment by gebart** [comment]
@LudwigOrtmann could you review this please?
RIOT will not build static libraries with `llvm-ar` without this change.

**Comment by LudwigOrtmann** [comment]
how do I obtain llvm-ar?

**Comment by gebart** [comment]
@LudwigOrtmann `llvm-ar` comes with the LLVM package on Gentoo (sys-devel/llvm) and Ubuntu (llvm-3.5)

**Comment by LudwigOrtmann** [comment]
Ah, sorry, I thought this was an OSX only issue...

**Comment by LudwigOrtmann** [comment]
(BTW: yes, apple apparently ships BSD ar.)

**Comment by LudwigOrtmann** [comment]
Works on current Arch Linux with/without llvm-ar and does not break building on OSX.

Please rebase.

**Comment by gebart** [comment]
Rebased, sorry for the confusion about OSX.

**Comment by LudwigOrtmann** [comment]
I just noticed that the commit message
```
Makefile.include: Refactor ar-wrapper test.
Hopefully should not be needed even on OSX when called with new arguments.
```
does not really reflect the change anymore, does it?

**Comment by gebart** [comment]
I will remove the comment in the commit message when I get back to my desktop. Everything fine otherwise?

**Comment by LudwigOrtmann** [comment]
I'd say so.

File: Makefile.include [hunk 1]
```diff
@@ -112,7 +112,9 @@ CFLAGS += -DCPU_$(CPUDEF)=\"$(CPU)\" -DRIOT_CPU=CPU_$(CPUDEF)
 CFLAGS += -DMCU_$(MCUDEF)=\"$(MCU)\" -DRIOT_MCU=MCU_$(MCUDEF)
 
 # OSX fails to create empty archives. Provide a wrapper to catch that error.
-ifneq (0, $(shell mkdir -p $(BINDIR); $(AR) -rc $(BINDIR)empty-archive.a 2> /dev/null; echo $$?))
+# llvm-ar does not understand `-rc`, but works for creating empty archives when
+# called with `rc` (without the leading dash).
```
Comment (LudwigOrtmann) [inline Makefile.include]: The comment is slightly superfluous as long as the commit message explains this. If you think it should be manifested somewhere, put it besides the `ARFLAGS` definition.

**Comment by LudwigOrtmann** [comment]
.. slipped in another remark while waiting for your return ;)

**Comment by gebart** [comment]
Reworded commit message and removed comment in Makefile.include. Good to go?

**Comment by LudwigOrtmann** [comment]
Yes, thanks =)

**Pull request merged**